PYME/IO/DataSources/BioformatsDataSource.py

```diff
@@ -25,7 +25,7 @@
     import bioformats
     #javabridge.start_vm(class_path=bioformats.JARS, run_headless=True)
 except:
-    pass
+    raise ImportError('Please make sure python-javabridge (https://pythonhosted.org/javabridge/) and python-bioformats (https://pythonhosted.org/python-bioformats/) are working before you use this module.')
```

Comment:
It would be good to know why these are in a `try ... pass` statement in the first place. I suspect there might be a reason for it (which may or may not be any good). There is a chance that throwing an import error here will break data loading more generally when bioformats is not present. More likely is that this was a hack to, e.g., get documentation generation running on a platform where bioformats wasn't installed. 

If going ahead with these changes I'd like to see the old code commented out, and a comment saying what had happened and that it was being removed as it wasn't obvious why.

Proposed alteration is not really that different to removing the `try ... pass` completely (which might be better as you get to see which module failed and potentially why).

Reply:
I don't think it will break loading more generally, as I haven't installed bioformats installed before today. We can drop the `try...pass`. Which old code do want commented out? I didn't delete anything besides `pass`, or is that what you're referring to?

Reply:
I was referring to the `pass` - just wanted to have a note in the code that indicated that this has changed without knowing why the old code was how it was when someone finds an error and comes looking. The circumstances where it would fail is if we import all the datasources somewhere (current code would ignore a failure to import bioformats). This might end up being in the task distribution stuff - there are places where we load data sources by module name (encoded in a string), rather than directly. These won't show up in a simple search of the source.